crates/net/network/src/manager.rs

```diff
@@ -108,6 +108,8 @@ pub struct NetworkManager<C> {
     metrics: NetworkMetrics,
     /// Disconnect metrics for the Network
     disconnect_metrics: DisconnectMetrics,
+    /// Transaction manager config
+    transactions_manager_config: TransactionsManagerConfig,
```

Comment:
we don't want to store this here. see other comment about moving the tx manager config to the network builder.

Reply:
if we don't store it in networkmanager, how should we access the field? isn't storing the config in networkbuilder too high level, since peers config is stored in swarm in networkmanager. should i do something like 
```
/// A builder that can configure all components of the network.
#[allow(missing_debug_implementations)]
pub struct NetworkBuilder<C, Tx, Eth> {
    pub(crate) network: NetworkManager<C>,
    pub(crate) transactions: Tx,
    pub(crate) request_handler: Eth,
    pub(crate) transactions_manager_config:  TransactionsManagerConfig,
}
```

Reply:
what about passing it as parameter to set transactions on the `NetworkBuilder`, is that possible?

Reply:
emm, ok, i can but feel convoluted since network manager takes in a config, so it should already have all the info unsure why we want to pass it as param which might lead an inconsistent case, where the NetworkManager config is diff from what got passed in. also unsure unsure how not store the transactionsmanagerconfig in networkconfig since i believe that's the result/output of networkbuilder, and builder's job seem finish once a networkconfig got produced, which passed around, copy relevant code here for ref.
```
 let config = NetworkConfigBuilder::new(secret_key)
            .disable_discovery()
            .listener_port(0)
            .build(client);
let (network_handle, network, mut transactions, _) = NetworkManager::new(config)
            .await
            .unwrap()
            .into_builder()
            .transactions(pool.clone())
            .split_with_handle(); 
```

Reply:
i push a commit that do it as param to transactions, but can;t get rid of it in networkconfig tho.